src/witan/app/handler.clj

```diff
@@ -27,24 +31,21 @@
 
 (defn login
   [body]
-  (let [username (:username body)
-        password (:password body)
-        valid? (user/user-valid? username password)]
-    (if valid?
+  (let [{:keys [username password]} body
+        valid-user (user/user-valid? username password)]
+    (if valid-user
       (let [token (user/random-token)]
         (swap! tokens assoc (keyword token) (keyword username))
-        (ok {:token token}))
+        (ok {:token token :id (:id valid-user)}))
       (ok {:message "login failed"}))))
 
 (defn signup
-  [body]
-  (let [username (:username body)
-        password (:password body)]
-    (if (user/add-user! username password)
-      (let [token (user/random-token)]
-        (swap! tokens assoc (keyword token) (keyword username))
-        (created {:token token}))
-      (ok {:message "User already present"}))))
+  [{:keys [username password name] :as body}]
+  (if-let [new-user (user/add-user! body)]
+    (let [token (user/random-token)]
+      (swap! tokens assoc (keyword token) (keyword username))
+      (created {:token token :id (:id new-user)}))
+    (ok {:message "User already present"})))
```

Comment:
aye, I thought of it as a temporary thing - better to check the email etc.
also, the user is logged in but doesn't have access to anything that is not 'public' as long as nobody shares stuff with them or adds them to groups.